src-cpp/rdkafkacpp.h

```diff
@@ -523,6 +523,78 @@ class KafkaConsumer;
 /**@endcond*/
 
 
+/**
+ * @name Error class
+ * @{
+ *
+ */
+
+/**
+ * @brief The Error class is used as a return value from APIs to propagate
+ *        an error. The error consists of an error code which is to be used
+ *        programatically, an error string for showing to the user,
+ *        and various error flags that can be used programmatically to decide
+ *        how to handle the error; e.g., should the operation be retried,
+ *        was it a fatal error, etc.
+ *
+ * Error objects must be deleted explicitly to free its resources.
+ */
+class RD_EXPORT Error {
+ public:
+
+ /**
+  * @brief Create error object.
+  */
+ static Error *create (ErrorCode code, const std::string *errstr);
+
+ virtual ~Error () { }
+
+ /*
+  * Error accessor methods
+  */
+
+ /**
+  * @returns the error code, e.g., RdKafka::ERR_UNKNOWN_MEMBER_ID.
+  */
+ virtual ErrorCode code () const = 0;
+
+ /**
+  * @returns the error code name, e.g, "ERR_UNKNOWN_MEMBER_ID".
+  */
+ virtual std::string name () const = 0;
+
+  /**
+   * @returns a human readable error string.
+   */
+ virtual std::string str () const = 0;
+
+ /**
+  * @returns true if the error is a fatal error, indicating that the client
+  *          instance is no longer usable, else false.
+  */
+ virtual bool is_fatal () const = 0;
+
+ /**
+  * @returns true if the operation may be retried, else false.
+  */
+ virtual bool is_retriable () const = 0;
+
+ /**
+  * @returns true if the error is an abortable transaction error in which case
+  *          the application may call RdKafka::Producer::abort_transaction()
+  *          and start a new transaction with
+  *          RdKafka::Producer::begin_transaction().
+  *          Else returns false.
```

Comment:
but what is the application developer expected to do if `false`?

Reply:
That is a very good question.
For some error types it indicates a bug in the app (e.g invalid arg), for others it might be a broker error code that we don't explicitly handle. 

What should the recommendation be? Always treat as fatal?

Reply:
i don't know that's why i'm looking for guidance :-). maybe document the possible causes, and suggest that treating as fatal is a safe thing to do?

Reply:
+1 for always fatal

Reply:
I updated the transactional API intro documentation

Reply:
I updated the transactional API documentation to include a recommendation how to handle errors that have neither retriable, abortable or fatal flag set, namely by treating them as fatal errors.

I'm hesitant to flag all these errors as fatal by default since the fatal errors are currently well defined in KIP-98 and the Java API .